definitions/features/instance.rb

```diff
@@ -44,4 +49,101 @@ def postgresql_local?
   def foreman_proxy_with_content?
     feature(:foreman_proxy) && feature(:foreman_proxy).with_content? && !feature(:katello)
   end
+
+  def ping?
+    if feature(:katello)
+      katello_ping
+    elsif external_proxy?
+      proxy_ping
+    else
+      foreman_ping
+    end
+    last_ping_result
```

Comment:
Why not passing the result as a return of `{katello,proxy,foreman}_ping`? Reads a bit weird this way

Reply:
Updated